File: SuperiorPackGroup/BLL/PalletTicketBLL.vb
Imports SuperiorPackGroup.SPGReportsTableAdapters

Public Class PalletTicketBLL

    Private m_PalletTicketTableAdapter As PalletTicketTableAdapter = Nothing

    Public ReadOnly Property Adapter() As PalletTicketTableAdapter
        Get
            If (Me.m_PalletTicketTableAdapter Is Nothing) Then
                Me.m_PalletTicketTableAdapter = New PalletTicketTableAdapter
                Me.m_PalletTicketTableAdapter.Connection.ConnectionString = My.Settings.UserConnectionString
            End If
            Return Me.m_PalletTicketTableAdapter
        End Get
    End Property

    Public Function GetPalletTicket(ByVal ItemID As Integer) As SPGReports.PalletTicketDataTable

        Return Adapter.GetPalletTicket(ItemID)

    End Function

End Class

File: SuperiorPackGroup/BLL/PalletTicketBLL.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using SuperiorPackGroup.SPGReportsTableAdapters;

namespace SuperiorPackGroup
{
	public class PalletTicketBLL
	{

		private PalletTicketTableAdapter m_PalletTicketTableAdapter = null;

		public PalletTicketTableAdapter Adapter
		{
			get
			{
				if (m_PalletTicketTableAdapter == null)
				{
					m_PalletTicketTableAdapter = new PalletTicketTableAdapter();
					m_PalletTicketTableAdapter.Connection.ConnectionString = Properties.Settings.Default.UserConnectionString;
				}
				return m_PalletTicketTableAdapter;
			}
		}

		public SPGReports.PalletTicketDataTable GetPalletTicket(int ItemID)
		{

			return Adapter.GetPalletTicket(ItemID);

		}

	}

}